pages/instructorFileEditor/instructorFileEditor.ejs

```diff
@@ -1,78 +1,193 @@
+                                    Doing so will pull from the remote GitHub repository and then will proceed to write your changes to disk, push them to the remote git repository, and sync them to the database.
+                                </div>
+                            </div>
+                            <div class="row my-4">
+                                <div class="col-auto">
+                                    <button name="__action" value="pull_and_save_and_sync" class="btn btn-primary">Retry save and sync after pull from remote GitHub repository</button>
+                                </div>
+                            </div>
+                            <% } %>
+                            <% if (fileEdit.jobSequenceId != null) { %>
+                            <div class="row collapse mt-4" id="results-<%= fileEdit.uuid %>">
+                                <div class="card card-body">
+                                    <%- include('../partials/jobSequenceResults', {job_sequence: fileEdit.jobSequence, job_sequence_enable_live_update: false}); %>
+                                </div>
+                            </div>
+                            <% } %>
+                        </div>
+                        <% } %>
 
-                <% if (fileEdit.didFindOutdated) { %>
-                    <div class="alert alert-warning alert-dismissible fade show" role="alert">
-                        The file you are trying to edit was changed by another user since your last saved draft. You must click "Revert to Original" to continue editing. Doing so will discard your saved draft. If you have made any changes that you would like to preserve, you should copy the text of your saved draft before proceeding.
-                        <button type="button" class="close" data-dismiss="alert" aria-label="Close">
-                            <span aria-hidden="true">&times;</span>
-                        </button>
+                        <% if ((fileEdit.alertChoice) && (!fileEdit.failedPush)) { %>
+                        <div class="alert alert-danger alert-dismissible fade show m-2" role="alert">
+                            <% if (fileEdit.hasSameHash) { %>
+                            You were editing this file and made changes. You may choose either to continue editing or to discard your changes.
+                            <% } else { %>
+                            Both you and another user made changes to this file. You may choose either to continue editing your draft or to discard your changes.
+                            <% } %>
+                            In particular, if you click <strong>Choose my version</strong> and then click <strong>Save and sync</strong>, you will overwrite the version of this file that is on disk. If you instead click <strong>Choose their version</strong>, any changes you have made to this file will be lost.
```

Comment:
Even after the user clicks "Choose my version" this error message is still displayed. Could we dismiss this alert after the user chooses a version? (Just this alert should be dismissed, I think, not the other alert saying that we failed to sync.)

Reply:
I think this alert does correctly vanish when the user clicks "Choose their version" because we reload the page.

Reply:
Done.